deployments/apiv2/services/users/src/main.py

```diff
@@ -126,9 +129,25 @@ async def login(request: Request, details: Union[UserLogin, CustomerLogin]):
                 ph.hash(pw)
                 raise LoginError(failed_msg)
             else:
-                scope = ["users:admin"] if is_customer_login_attempt else json.loads(row.get("scope", "[]"))
-                return await _create_new_tokens(con, row["id"], customer_id, scope, account_type)
-
+                # both users and customers have scopes
+                # check account usage quotas
+                cust_id = customer_id if customer_id is not None else row["id"]
+                usage_quota = await check_customer_quota(con, str(cust_id), details.service)
+                scope = json.loads(row.get("scope", "[]"))
+
+                if is_customer_login_attempt:
+                    # get tier of service scope in list of customer scopes
+                    scope = [s for s in scope if details.service in s]
+                    # replace with customer scope
+                    # raises IndexError if not found
+                    scope[0] = scope[0].replace(details.service, "customer")
+
+                tokens = await _create_new_tokens(con, row["id"], customer_id, scope, account_type)
+                return LoginResponse(tokens=tokens, usage_quota=usage_quota)
+    except IndexError:
```

Comment:
I think it would be better to catch the index error right where it happens and then raise a login error with this message instead, then it can just get handled by `except LoginError as e:`

Reply:
Or you could just do
```if not scope:
   raise LoginError(msg)
```